code/modules/paperwork/paper.dm

```diff
@@ -454,6 +454,20 @@
 	ui_interact(user)
 	return ..()
 
+/// Secondary right click interaction to quickly stamp things
+/obj/item/paper/attackby_secondary(obj/item/weapon, mob/user, params)
+	var/writing_stats = weapon.get_writing_implement_details()
+
+	if(writing_stats["interaction_mode"] == MODE_STAMPING)
+		if(!user.can_read(src) || user.is_blind()) // Just leftclick instead
+			return
+		else
+			add_stamp(writing_stats["stamp_class"], rand(0, 400), rand(0, 500), stamp_icon_state = writing_stats["stamp_icon_state"])
```

Comment:
What are 400 & 500 exactly? Get the dimensions of the paper and replace them here